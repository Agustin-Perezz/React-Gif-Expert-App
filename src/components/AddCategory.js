import React, { useState } from 'react'
import { PropTypes } from "prop-types";

export const AddCategory = ( { setCategorias } ) => {

	const [inputValue, setInputValue] = useState('');

	const handleEventChange = (e) => {
		setInputValue( e.target.value );
	}

	const handleSubmit = (e) => {
		e.preventDefault();
		if ( inputValue.trim().length > 2 ) {
			setCategorias( cats => [  inputValue, ...cats, ]);
			setInputValue( '' );
		}

	}

	return (
		<form onSubmit={ handleSubmit }>
			<input 
				value={inputValue}
				type="text" 
				onChange={ handleEventChange }
			/>
		</form>
	)
}

AddCategory.propTypes = {
	setCategorias: PropTypes.func.isRequired
}
