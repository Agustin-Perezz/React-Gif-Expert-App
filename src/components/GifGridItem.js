import PropTypes from 'prop-types'

export const GifGridItem = ( { url, id, title } ) => {

	return (
		<div className="card animate__animated animate__fadeIn">
			<img src={ url } alt= { title } />
			<div className="title">
				<p> { title } </p>
			</div>
		</div>
	)
}

GifGridItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired
}