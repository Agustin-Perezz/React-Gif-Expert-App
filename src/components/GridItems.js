import { GifGridItem } from './GifGridItem'
import { useFetchApi } from "../hooks/useFetchApi"

export const GridItems = ( { category } ) => {

	// const [images, setImages] = useState([]);

	const { loading, data:images } = useFetchApi( category );
	console.log( images )


	return (
		<>
			<h3> { category } </h3>	
			{ loading && 'Loading..'}
			<div className="card-grid">

				{
					images.map( (img) => {
						return <GifGridItem 
							key= { img.id }
							{ ...img }
						/>
					})
				}

			</div> 
		</>
	)
}
