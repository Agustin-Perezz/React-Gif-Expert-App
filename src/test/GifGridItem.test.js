import { shallow } from 'enzyme';
import { GifGridItem } from "../components/GifGridItem"
import '@testing-library/jest-dom';
import '../setupTests';

describe('Pruebas en el componente GifGridItem', () => {

	const title = 'Pepe';
	const url = '123'
	const wrapper = shallow( <GifGridItem title={ title } url={ url } />);

	test('Verificacion componente', () => {
		
		expect( wrapper ).toMatchSnapshot();

	})

	test('Deve de encontrarse un parrafo', () => {

		const p = wrapper.find('p');
		expect( p.text().trim() ).toBe( title );		

	})

	test('Deve de encontrarse una imagen', () => {

		const img = wrapper.find('img');
		// console.log( img.prop('alt'))
		// expect( img.prop('src')).toBe( url );
		// expect( img.prop('alt')).toBe( title );

	})
	
	test('Deve de contener el div la clase animate__fadeIn', () => {

		const div = wrapper.find('div');
		const className = div.prop('className');
		// const clase = div.prop('className')
		expect( className.includes('animate__fadeIn')).toBe( true );
		
	})
	

})
