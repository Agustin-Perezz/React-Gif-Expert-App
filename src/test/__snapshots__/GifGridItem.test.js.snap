// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGridItem Verificacion componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Pepe"
    src="123"
  />
  <div
    className="title"
  >
    <p>
       
      Pepe
       
    </p>
  </div>
</div>
`;
